regcomp.c

```diff
@@ -12745,6 +12745,16 @@ S_regpiece(pTHX_ RExC_state_t *pRExC_state, I32 *flagp, U32 depth)
         if (!(flags&HASWIDTH) && op != '?')
           vFAIL("Regexp *+ operand could be empty");
 #endif
+
+        /* There's no point in trying to match something 0 length more than
+         * once except for extra side effects, which we don't have here since
+         * not POSTPONED */
```

Comment:
@demerphq made this die for \K shortly before I was working on this code.  My sense was that he felt pretty strongly that no one should write code that had a >1 quantifier in \K.   I figure he knew better than I, and that I should change the existing behavior of the other cases, or that's my recollection.    Those actions are taken only for high maxes      `if (max > REG_INFTY/3) {` whereas there still is no point in doing it more than once even if we don't make a stink about it.  I left in the warning at that value to avoid changing any behavior.

Reply:
@demerphq never justified that despite requests; I feel pretty strongly that it should _not_ die.

Reply:
I agree that it's better not to die

Reply:
But when I tried it, I found that other things need to change.  I think then that this is outside the scope of this PR, so I am deferring it.